.github/workflows/build.yml

```diff
@@ -114,3 +114,37 @@ jobs:
       - run: ./tool/shared/deploy.sh --robots ok default
         env:
           FIREBASE_TOKEN: ${{ secrets.FIREBASE_TOKEN }}
+
+  deploy_preview:
```

Comment:
It would be possible to avoid the duplication by merging the if: expression here and then have the specific ifs further down in the deployment steps. The ifs make it harder to read the workflow, so I'm also OK with it as is.

Reply:
> I think that would mean adding if's to each and every step here, as we don't want to run the build, etc. unless we're doing a push to master or a PR from a non-fork?

(copied comment from the other thread)

No:
"if master or PR" on `deploy:`
"if master" on the step needed post submit.
"if PR" on the step needed for PRs.

The remaining steps are the same.